libraries/ms-helper/test/test_sequenced_relay.c

```diff
@@ -0,0 +1,135 @@
+#include "can.h"
+#include "can_transmit.h"
+#include "delay.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "sequenced_relay.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_SEQUENCED_RELAY_NUM_RX_HANDLERS 5
+#define TEST_SEQUENCED_RELAY_CAN_DEVICE_ID 0x1
+
+#define TEST_SEQUENCED_RELAY_LEFT \
+  { .port = GPIO_PORT_B, .pin = 3 }
+#define TEST_SEQUENCED_RELAY_RIGHT \
+  { .port = GPIO_PORT_B, .pin = 9 }
+#define TEST_SEQUENCED_RELAY_DELAY_MS 10
+
+static CANStorage s_can;
+static CANRxHandler s_rx_handlers[TEST_SEQUENCED_RELAY_NUM_RX_HANDLERS];
+static SequencedRelayStorage s_sequenced_relay;
+
+typedef enum {
+  TEST_SEQUENCED_RELAY_EVENT_CAN_RX = 0,
+  TEST_SEQUENCED_RELAY_EVENT_CAN_TX,
+  TEST_SEQUENCED_RELAY_EVENT_CAN_FAULT,
+} TestSequencedRelayEvent;
+
+static StatusCode prv_ack_cb(CANMessageID msg_id, uint16_t device, CANAckStatus status,
```

Comment:
We should probably make a helper for this. I've used this paradigm at least a half dozen times. A test function called `test_can_ack_with_context` or something.

Reply:
Ah you seem to test for different things though